File: BackEnd/Comun/Mozo.Model/Maestro/ArchivoModel.cs
namespace Mozo.Model.Maestro;

public record ArchivoFilterDto : BaseFilterDto
{
    public int? CoEmpresa { get; set; }
    public int? CoArchivo { get; set; }
    public int? CoTipoEntidad { get; set; }
    public int? CoEntidad { get; set; }
    public int? CoTipo { get; set; }
    public int? NuOrden { get; set; }


}





public class OrdenArchivoDto
{
    public int CoArchivo { get; set; }
    public int NuOrden { get; set; }
}

public class OrdenMasivoRequest
{
    public int FlEmpresaNotKey { get; set; }
    public int? CoEmpresa { get; set; }
    public List<OrdenArchivoDto> Ordenes { get; set; } = new();
}


[Serializable]
public partial class ArchivoModel : BaseModel //<ArchivoModel>
{

    public int? CoEmpresa { get; set; }
    public int? CoTipoEntidad { get; set; }
    public int? CoEntidad { get; set; }
    public int? CoArchivo { get; set; }
    public int? CoTipo { get; set; }

    public string? NoArchivo { get; set; }
    public string? NoExtension { get; set; }
    public string? NoRuta { get; set; }

    public int? NuOrden { get; set; }
    public string? TxDescripcion { get; set; }
    public int? FlDefault { get; set; }
    public long? NuBytes { get; set; }
    public int? NuAlto { get; set; }
    public int? NuAncho { get; set; }
    public string? NoTitulo { get; set; }

}

public partial class ArchivoModel
{
    public string? Url { get; set; }
    public string? ContentType { get; set; }
    public int? FlGaleria { get; set; }
}

File: FrontEnd-Admin/mozo-app/src/app/shared/models/maestro/archivo.model.ts
import { BaseModel } from "@sharedModels/base.model";

export interface ArchivoModel extends BaseModel {
  coEmpresa?: number | null;
  coArchivo: number;
  coTipoEntidad: number;
  coEntidad: number;
  coTipo: number;

  noArchivo?: string;
  noExtension?: string;
  noRuta?: string;
  nuBytes: number;
  nuAlto?: number | null;
  nuAncho?: number | null;

  nuOrden: number;
  txDescripcion?: string;
  flDefault: number;
  noTitulo?: string;

  // Calculados desde el backend
  url?: string;
  contentType?: string;
}

// ===== INTERFACES =====

export interface ArchivoUploadParams {
  flEmpresaNotKey?: number;
  coEmpresa?: number | null;
  coTipoEntidad: number;
  coEntidad: number;
  coTipo: number;
  nuOrden?: number;
  txDescripcion?: string;
  noTitulo?: string;
  flDefault?: number;
  flGaleria:number;
}

export interface UploadProgress {
  progreso: number;
  completado: boolean;
  archivo?: ArchivoModel;
  ids?: number[];
  fase?: 'comprimiendo' | 'subiendo' | 'finalizado';
}

export interface DownloadUrlOptions {
  width?: number;
  format?: 'webp' | 'jpg' | 'png';
  coEmpresa?: number;
}

export interface ArchivoFilterDto {
  flEmpresaNotKey?: number;
  coEmpresa?: number | null;
  coArchivo?: number;
  coTipoEntidad?: number;
  coEntidad?: number;
  coTipo?: number;
  nuOrden?: number;
  flGaleria?: number;
}

export interface OrdenArchivoDto {
  coArchivo: number;
  nuOrden: number;
}

export interface OrdenMasivoRequest {
  flEmpresaNotKey: number;
  coEmpresa?: number | null;
  ordenes: OrdenArchivoDto[];
}

